interfaces/datastructure/Frame.h

```diff
@@ -25,20 +25,20 @@ class SOLARFRAMEWORK_API Frame {
     ///
 
 public:
-    Frame(const SRef<Frame> frame);
+    explicit Frame(const SRef<Frame>& frame);
 	
-	Frame(const SRef<Keyframe> keyframe);
+	explicit Frame(const SRef<Keyframe>& keyframe);
 
-    Frame(const std::vector<Keypoint> & keypoints,
-          const SRef<DescriptorBuffer> descriptors,
-          const SRef<Image> view,
+    Frame(std::vector<Keypoint> keypoints,
```

Comment:
We must copy keypoints as they can be used later in the pipeline, hence move semantics in this case will leave the calling vector undefined.